csrc/device_lower/pass/index.cpp

```diff
@@ -1816,6 +1821,230 @@ Val* hardCodedIndexGenerationForStMatrix(
+//    *       *       *       *       *
+// 32 *********************************
+//    *       *       *       *       *
+//    *       *       *       *       *
+// W2 *  I0   *  I1   *  I2   *   I3  *
+//    *       *       *       *       *
+//    *       *       *       *       *
+// 48 *********************************
+//    *       *       *       *       *
+//    *       *       *       *       *
+// W3 *  I0   *  I1   *  I2   *   I3  *
+//    *       *       *       *       *
+//    *       *       *       *       *
+// 64 *********************************
+//
+// Calculate row in the TMA Box [M(64), NI(64)]:
+//   row_for_the_warp = warp_id * stsm_m_tile(16)
+//   row_for_the_lane = lane_id % stsm_m_tile(16)
+//   row = row_for_the_warp + row_for_the_lane
+//
+// Calculate column in the TMA Box [M(64), NI(64)]:
+//   column_for_the_lane = lane_id / stsm_n_tile(16)
+//   number_of_columns_per_stsm_x4 = stsm_n_tile(16) / stsm_column_size(8)
+//   column_for_the_nio_loop = inner_index * number_of_columns_per_stsm_x4
+//   column = column_for_the_lane + column_for_the_nio_loop
+//
+// The 128B swizzle is applied to a (8, 64) matrix of dtype fp16 or bf16.
+// The  size of fp16 and bf16 is 2B. The 64 elements along the inner dimension
+// contain 128B, which fill all 32 4B shared memory banks. Contiguous sections
+// of 8 elements are grouped together into 16B megabanks. The 16B megabank
+// corresponds with the 128-bit vectorized load. The 8 megabanks are swizzled
+// with the 8 rows of the matrix to avoid bank conflicts. This swizzle pattern
+// is repeated along the rows of the TMA box.
+//
+// Swizzle column
+//   row_in_swizzle_pattern = row % swizzle_row_size(8)
+//   swizzle_col = column XOR row_in_swizzle_pattern
+//
+// Calculate Tile Offset
+//   row_offset = row * NI(64) * 2(half)
+//   column_offset = column * swizzle_size(8) * 2(half)
+//   tile_offset = row_offset + column_offset
+//
+// Get shared memory offset
+//   smem_offset = offset_from_tdy + offset_from_outer_index + tile_offset
+Val* hardCodedIndexGenerationForStMatrix128BSwizzle(
+    const LoadStoreOp* ldst,
+    ForLoop* loop,
+    const int64_t stsm_m_tile,
+    const int64_t stsm_n_tile,
+    const int64_t tma_m,
+    const int64_t tma_n) {
+  NVF_ERROR(
+      (stsm_m_tile == 8 && stsm_n_tile == 8) ||
+          (stsm_m_tile == 16 && stsm_n_tile == 8) ||
+          (stsm_m_tile == 16 && stsm_n_tile == 16),
+      "size not currently supported for stmatrix");
+
+  NVF_ERROR(
+      ldst->out()->dtype() == DataType::Half,
+      "we only support half type in stmatrix");
+
+  NVF_ERROR(ldst->out()->isA<TensorView>());
+  TensorView* out_tv = ldst->out()->as<TensorView>();
+  NVF_ERROR(getSwizzle(out_tv) == MmaInputSmemSwizzle::B128);
+
+  // Constants
+  constexpr int64_t dtype_size = 2;
+  constexpr int64_t warp_size = 32;
+  constexpr int64_t swizzle_row_size = 8;
+  constexpr int64_t stsm_column_size = 8;
+  constexpr int64_t swizzle_n_tile = 64;
```

Comment:
For e.g. a 64B swizzle, is the only thing that changes this value? I know it's probably not but I am trying to gauge how much of this we'll be able to reuse for the other swizzles.

Reply:
`swizzle_row_size` would be 4 for `64B swizzle`.
`swizzle_n_tile` would be `32` so `swizzle_n_iter` would be 2.

Need make this change: `swizzle_n_iter = swizzle_n_tile / stsm_n_tile`

Probably need to double check all `swizzle_row_size` and `stsm_column_size`.